src/main/java/com/amazonaws/services/sqs/AmazonSQSIdleQueueDeletingClient.java

```diff
@@ -189,17 +189,26 @@ public CreateQueueResult createQueue(CreateQueueRequest request) {
         if (!queueName.startsWith(queueNamePrefix)) {
             throw new IllegalArgumentException();
         }
-
+        String retentionPeriodString = retentionPeriod.get().toString();
+        int retry = 0;
         CreateQueueRequest superRequest = request.clone()
                 .withQueueName(queueName)
-                .withAttributes(attributes);
+                .withAttributes(attributes)
+                .addTagsEntry(IDLE_QUEUE_RETENTION_PERIOD_TAG, retentionPeriodString);
 
         CreateQueueResult result = super.createQueue(superRequest);
         String queueUrl = result.getQueueUrl();
 
-        String retentionPeriodString = retentionPeriod.get().toString();
-        amazonSqsToBeExtended.tagQueue(queueUrl,
-                Collections.singletonMap(IDLE_QUEUE_RETENTION_PERIOD_TAG, retentionPeriodString));
+        // In case of throttling issues retry tagging the queue
+        while (retry++ < 5 && !listQueueTags(queueUrl).getTags().containsKey(IDLE_QUEUE_RETENTION_PERIOD_TAG)) {
+            try {
+                TimeUnit.SECONDS.sleep(1);
+                amazonSqsToBeExtended.tagQueue(queueUrl,
+                        Collections.singletonMap(IDLE_QUEUE_RETENTION_PERIOD_TAG, retentionPeriodString));
+            } catch (InterruptedException e) {
+                e.printStackTrace();
```

Comment:
Do not swallow exceptions! Either re-throw this or (for this particular kind of exception) invoke `Thread.currentThread().interrupt()` to reset the interrupt flag.